src/command.rs

```diff
@@ -49,6 +49,42 @@ macro_rules! delegate {
     }};
 }
 
+/// If a command is `OverSsh` then it can be executed over an SSH session.
+/// Primarily a way to allow `std::process::Command` to be turned directly into an `openssh::Command`.
+pub trait OverSsh {
+    /// Given a session, return a command that can be executed over that session.
+    fn over_session<'session>(&self, session: &'session Session) -> crate::Command<'session>;
+}
+
+impl OverSsh for std::process::Command {
+    /// Given a session, convert a std::process::Command into an `openssh::Command` 
+    /// that can be executed over that session.
+    /// ```no_run
+    /// use std::process::Command;
+    /// use openssh::{Session, KnownHosts, OverSsh};
+
+    /// let session = Session::connect_mux("me@ssh.example.com", KnownHosts::Strict).await?;
+    /// let ls = 
+    ///     Command::new("ls")
+    ///     .arg("-l")
+    ///     .arg("-a")
+    ///     .arg("-h")
+    ///     .over_session(&session)
+    ///     .output()
+    ///     .await?;
+    /// 
+    /// assert!(String::from_utf8(ls.stdout).unwrap().contains("total"));
+    /// 
+    /// 
+    /// ```
+    fn over_session<'session>(&self, session: &'session Session) -> Command<'session> {
+
+        let mut command = session.raw_command(self.get_program());
```

Comment:
I think we should use `Session::command` here to escape the program, since the ssh protocol essentially concats program and args into one string `$program $arg0 $arg1 ...` and pass it to shell.

Reply:
hmm I'm not sure I fully understand this. As it stands, `self.get_program()` returns an `&OsStr` but `Session::command` expects `Into<Cow<'a, str>>`. A meaningful way to get that is to `self.get_program().to_string_lossy()` but I'm a little worried that changes the semantics here. please lmk if I misunderstood it.

Reply:
Oh I didn't realize the difference between typing.

Yeah that's certainly an issue, but I still think it's necessary to escape them, e.g. if the program contains `'` or `!`, then it will be interpreted by the shell instead of as part of the program name.

We could use [`OsStrExt::as_bytes`](https://doc.rust-lang.org/std/os/unix/ffi/trait.OsStrExt.html#tymethod.as_bytes) then escapes them in the same way as [`shell_escape::unix::escape`](https://docs.rs/shell-escape/latest/src/shell_escape/lib.rs.html#101-120).

Reply:
There are still some typing incompatibilities as `shell_escape::unix::escape` expects a `Cow<str>`. I tried to implement a variant for `Cow<OsString> -> Cow<OsString>` but I couldn't find a way to interleave special (i.e. `[^\!']`) characters with the non-special ones. 

Hoping you could suggest a fix for what you have in mind.

Reply:
You can use [`OsStrExt::as_bytes`](https://doc.rust-lang.org/std/os/unix/ffi/trait.OsStrExt.html#tymethod.as_bytes) to convert `OsStr` to `&[u8]` first, then you can implement [`shell_escape::unix::escape`](https://docs.rs/shell-escape/latest/src/shell_escape/lib.rs.html#101-120) but for bytes `&[u8]` instead of `&str`.

Reply:
I just recall there is plan to support windows, but it will take probably at least another 3-6 months before it is even ready for review (I am working some other projects right now).

So this is probably fine.
For windows, we would have to use some different implementations.

Reply:
that makes sense. lets go with this for now.